lib/secret-storage/nondumpable-allocator.c

```diff
@@ -116,11 +117,23 @@ nondumpable_buffer_alloc(gsize len)
   return buffer->user_data;
 }
 
+static void
+nondumpable_mem_zero(gpointer s, gsize len)
+{
+#if SYSLOG_NG_HAVE_EXPLICIT_BZERO
+  explicit_bzero(s, len);
+#else
+  typedef void *(*MemsetFunc)(void *, gint, gsize);
+  static volatile MemsetFunc memset_func = memset;
+  memset_func(s, 0, len);
+#endif
+}
```

Comment:
If this is not a performance critical path, then why don't we simply iterate on it and set all bytes to zero?

Reply:
We can iterate on it, but it has to be at least volatile to prevent the mentioned optimization.
@algernon has az excellent article about this in the previous comment.